libs/auth/src/apdu.ts

```diff
@@ -286,12 +286,24 @@ export function constructTlv(
 }
 
 /**
- * The inverse of constructTlv, splits a TLV into its tag, length, and value
+ * A Tag-Length-Value (TLV) structure.
+ *
+ * See https://en.wikipedia.org/wiki/Type-length-value for more information.
+ */
+export interface Tlv {
+  tag: Buffer;
+  length: Buffer;
+  value: Buffer;
+}
+
+/**
+ * Parses a TLV from a buffer, returning the remainder of the buffer and the parsed TLV. If the tag
+ * does not match the expected tag, an error is thrown.
  */
-export function parseTlv(
+export function parseTlvPartial(
   tagAsByteOrBuffer: Byte | Buffer,
   tlv: Buffer
-): [tag: Buffer, length: Buffer, value: Buffer] {
+): [remainder: Buffer, tlv: Tlv] {
```

Comment:
nit: Very minor preference for putting the remainder after the tlv in the return value as that feels a bit more intuitive than having the remainder come first

Reply:
4df01ada191004de047ec78e3d04998569db33e1